### DISCUSSION TITLE
params or state not working

### DISCUSSION THREAD

**Discussion by macingili** [vuejs/router#1857] [Help and Questions] [answered]
Hi
This code is not working. is there a problem or am i doing it wrong ?

router/ndex.js
 const router = createRouter({
    history: createWebHashHistory(),
    routes
});
export default router;

<router-link :to="{ name: 'user', params: { username: 'erina' }}">
                        user
                      </router-link>

or

 <Button label="xx" @clik="$router.push({name: 'user', params: {id: 'xx'}})" />

 or 

<router-link :to="{ name: 'user', state: { username: 'erina' }}">
                        user
                      </router-link>

or

 <Button label="user" @clik="$router.push({name: 'user', state: {id: 'xx'}})" />

if i use params it warns that it is removed. If I use state it doesn't warn but it also doesn't work.

**Comment by posva** [comment, marked as answer]
state can be found in `window.history.state` (client only)
